doc/user/load_save.md

```diff
@@ -1,61 +1,563 @@
+| `PPM`                     | `opts.Format() = FileType::PPMBinary;`  | [Numeric](#numeric-data) data. | Load/save in the PPM image format; data should have values in the range `[0, 255]`.  The size of the image will be the same as the size of the matrix (after any transpose is applied). |
+| `HDF5`                    | `opts.Format() = FileType::HDF5Binary;` | [Numeric](#numeric-data) data. | Load/save in the [HDF5](https://en.wikipedia.org/wiki/Hierarchical_Data_Format) binary format; only available if Armadillo is configured with [HDF5 support](https://arma.sourceforge.net/docs.html#config_hpp). |
+| `ArmaBin`                 | `opts.Format() = FileType::ArmaBinary;` | [Numeric](#numeric-data) data. | Load/save in the space-efficient [`arma_binary`](https://arma.sourceforge.net/docs.html#save_load_mat) format (packed binary data). |
+| `RawBinary`               | `opts.Format() = FileType::RawBinary;`  | [Numeric](#numeric-data) data. | Load/save as packed binary data with no header and no size information; data will be loaded as a single column vector _(not recommended)_. |
+|---------------------------|-----------------------------|---------------------------|-------------------|
+
+### `TextOptions`
+
+The `TextOptions` class represents options specific to matrix types stored in
+plaintext formats ([numeric](#numeric-data) and [categorical](#mixed-categorical-data)
+data).  `TextOptions` is a child class and thus any standalone operators or members from its parent classes are also available:
+
+ * [`DataOptions`](#dataoptions) provides:
+   - `Fatal`, `NoFatal`, and `AutoDetect` standalone operators
+   - `opts.Fatal()` and `opts.Format()` members
+   - See the [`DataOptions` operator and member documentation](#dataoptions-standalone-operators-and-members)
+ * [`MatrixOptions`](#matrixoptions) provides:
+   - `Transpose`, `NoTranspose`, `PGM`, `PPM`, `HDF5`, `ArmaBin`, and
+     `RawBinary` standalone operators
+   - `opts.Transpose()` member
+   - See the [`MatrixOptions` operator and member documentation](#matrixoptions-standalone-operators-and-members)
+
+#### `TextOptions` standalone operators and members
+
+The options below can be used as standalone operators to the
+[`data::Load()`](#dataload) and [`data::Save()`](#datasave) functions, or as
+calls to set members of an instantiated `TextOptions` object.
+
+If an option is given that does not match the type of data being loaded or
+saved, if [`Fatal()`](#dataoptions-standalone-operators-and-members) is set,
+then an exception will be thrown; otherwise, a warning will be printed if
+[`MLPACK_PRINT_WARN`](compile.md#configuring-mlpack-with-compile-time-definitions)
+is set.
+
+<!-- TODO for Omar: what happens when HasHeaders() is true for saving? -->
```

Comment:
Can you check into this one?  Do we save CSV headers if the user specifies `HasHeaders()` and `Headers()` before saving?

Reply:
Agreed, or two separate PRs for that, totally up to you.